docs/ADR-51-catalyst-content-validations.md

```diff
@@ -0,0 +1,336 @@
+# Catalyst Conent validations
+
+## Context and Problem Statement
+
+Catalysts receive entity deployments that will be stored and distributed among them. Some validations are executed for each entity deployment, depending on the context and entity type.
+
+Most validations need information about the Catalyst content state and external dependencies like The Graph, which makes them very hard to predict, i.e. we can't ensure an entity deployment will be correct without validating it in a given context.
+
+On the other hand, we have some stateless validations that may be useful to run before actually doing a deployment, like metadata schema and hashes validations.
+
+Adding an entity type to a Catalyst requires a tedious effort which envolves defining schemas, relations and validations in different repositories.
+
+For instance, adding a validation to the Catalyst can be overwhelming for a developer from outside the Platform team.
+
+## Decision
+
+Validations will be moved to a library that will not depend on any other than definitions like schemas and types.
+
+It will expose an interface that will be implemented and required for validations execution, at the moment it means the Catalyst will provide this implementation but any other could do it.
+
+We will redefine the deployment contexts and only take into account: 'LOCAL' and 'SYNCED'.
+
+All validations will know this context and will decide if must do something different, but no validation list per context will exist, i.e. all validations will run on each deployment.
+
+Also, we will get rid of the entity version idea, moving to a timestamp based decision making on validations, so we can avoid duplicated/deprecation logic. To do so, we will be using timestamp defined in (ADR-45)[1].
+
+### Validations
+
+We will execute these validations for all entity types:
+
+- SIGNATURE
+
+  Validate that the signature belongs to the Ethereum address
+
+  ([reference](https://github.com/decentraland/decentraland-crypto/blob/master/src/Authenticator.ts))
+
+```
+function validateSignature: (see reference)
+  1. Fail if authChain is malformed
+  2. For each authLink L in authChain
+    a. Let validator V be the validator by L.type
+    b. Fail if V can't validate on that time
+
+
+1. Let validateSignature be a function that validates the auth chain of the deployment
+
+2. Fail if validateSignature(entity.id, deployment.authChain, deployment.timestamp)
+```
+
+- IPFS HASHING
+
+  Validates that all hashes used by the entity are actually IPFS hashes
+
+```
+parameter ADR_45_TIMESTAMP = 1648771200
+
+1. If deployment.timestamp <= ADR_45_TIMESTAMP return
```

Comment:
I think the IPFS_HASH_V0 validation is missing here

Reply:
We were talking about using only V1 after this time, is that incorrect?